app/models/internal/reports/Recipient.java

```diff
@@ -16,62 +16,125 @@
 
 package models.internal.reports;
 
-import models.internal.impl.DefaultEmailRecipient;
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.arpnetworking.metrics.portal.reports.RecipientType;
+import net.sf.oval.constraint.Email;
+import net.sf.oval.constraint.NotNull;
 
+import java.util.Objects;
 import java.util.UUID;
 
 /**
  * Internal model for a recipient, i.e. a destination.
  *
  * @author Christian Briones (cbriones at dropbox dot com)
  */
-public interface Recipient {
+public final class Recipient {
```

Comment:
I think the other internal models still have an interface and then a `DefaultFoo` implementation.